code/game/objects/items/blueprints.dm

```diff
@@ -238,19 +238,17 @@
 		return
 
 	//stuff tied to the area to rename
-	var/list/to_rename = list(
+	var/static/list/to_rename = typecacheof(list(
 		/obj/machinery/airalarm,
 		/obj/machinery/atmospherics/components/unary/vent_scrubber,
 		/obj/machinery/atmospherics/components/unary/vent_pump,
 		/obj/machinery/door,
 		/obj/machinery/firealarm,
 		/obj/machinery/light_switch,
 		/obj/machinery/power/apc,
-	)
-
-	for(var/turf/area_turf as anything in area.get_contained_turfs())
-		for(var/obj/machine in area_turf)
-			if(!is_type_in_list(machine, to_rename))
-				continue
-			machine.name = replacetext(machine.name, oldtitle, title)
+	))
+	for (var/list/zlevel_turfs as anything in area.get_zlevel_turf_lists())
+		for (var/turf/area_turf as anything in zlevel_turfs)
+			for(var/obj/machine as anything in typecache_filter_list(area_turf.contents, to_rename))
```

Comment:
is this slower? since the other case filters by object first whereas this one means we have to feed the whole of contents into the filter (and byond filtering is almost always faster)

Reply:
typecache is generally faster then istype or byond filtering if you are filtering by 3 types or more. its hard to know how that fairs with a machine byond pre-filter, as well as byond filtering out mobs because it was an object type path, but i'll unstage those changes and we can a|b pr it later to see.

Reply:
I'm pretty sure byond filtering is faster then istype, tho I would like to know for sure

Reply:
well ya, but typecache filter list doesn't use istype()